CONTRIBUTING.md

```diff
@@ -189,11 +189,13 @@ Introducing new Parse Errors requires the following steps:
 Introducing new [Parse Server configuration][config] parameters requires the following steps:
 
 1. Add parameters definitions in [/src/Options/index.js][config-index].
-1. If a nested configuration object has been added, add the environment variable option prefix to `getENVPrefix` in [/resources/buildConfigDefinition.js](https://github.com/parse-community/parse-server/blob/master/resources/buildConfigDefinition.js).
-1. Execute `npm run definitions` to automatically create the definitions in [/src/Options/Definitions.js][config-def] and [/src/Options/docs.js][config-docs].
-1. Add parameter value validation in [/src/Config.js](https://github.com/parse-community/parse-server/blob/master/src/Config.js).
-1. Add test cases to ensure the correct parameter value validation. Parse Server throws an error at launch if an invalid value is set for any configuration parameter.
-1. Execute `npm run docs` to generate the documentation in the `/out` directory. Take a look at the documentation whether the description and formatting of the newly introduced parameters is satisfactory.
+2. If a parameter group (nested parameter) has been added:
+   - add the environment variable prefix for the parameter group to `nestedOptionEnvPrefix` in [/resources/buildConfigDefinition.js](https://github.com/parse-community/parse-server/blob/master/resources/buildConfigDefinition.js)
+   - add the parameter group type to `nestedOptionTypes` in [/resources/buildConfigDefinition.js](https://github.com/parse-community/parse-server/blob/master/resources/buildConfigDefinition.js)
+3. Execute `npm run definitions` to automatically create the definitions in [/src/Options/Definitions.js][config-def] and [/src/Options/docs.js][config-docs].
```

Comment:
parameter group / group type nested parameters sounds confusing. I think we should make the user aware of the `/resources/buildConfigDefinition.js` file. Like if you need to add definition types add them to `buildConfigDefinition.js`

I want to add a new type `NumberOrObject` should we handhold the developers?

Reply:
> parameter group / group type / nested parameters sounds confusing.

Yes, I struggled myself to find a proper term. A new config parameter only needs to be added to `buildConfigDefinition.js` if it is a new interface type, otherwise the file doesn't have to be touched.

Reply:
To be honest, when I first documented the definitions, I assumed the `buildConfigDefinition.js` would rarely get updated (4 people have touched it in 4 years).

@davimacedo Since you haven't touched the `buildConfigDefinition.js` do you have any suggestions?

Reply:
Maybe we can automate the 2 things that sometimes need to get updated.

Reply:
Any update on this?

Reply:
@dplewis I reworded this with an example of an existing parameter. That should make it easier to follow.